### ISSUE TITLE
Add `deriveEq` or analog

### ISSUE THREAD

**Issue by uhbif19** [IntersectMBO/plutus#5129]
### Describe the feature you'd like

I saw `deriveEq` in examples and it was mentioned in issues, but I cannot find it in github or documentation. What happened to it?

### Describe alternatives you've considered

Writing `Eq` instances by hand.

**Comment by michaelpj** [comment]
There was never a `deriveEq` in this repository. I think maybe someone else wrote one?

**Comment by uhbif19** [comment]
But it is mentioned in your issue :thinking:  https://github.com/input-output-hk/plutus/issues/4537

Maybe we mixed up and there was no such functionality.

Will adding `deriveEq` or its Generic equivalent considered a good PR for this repo?

**Comment by michaelpj** [comment]
It is mentioned in the example given by the issue creator, I assume they wrote one or got it from somewhere.

> Will adding deriveEq or its Generic equivalent considered a good PR for this repo?

If it's not too much code then probably yes.

**Comment by uhbif19** [comment]
Sorry, you right it probably was code from some closed repo.

**Comment by effectfully** [comment]
@uhbif19 would you be interested in helping us out?

Maybe it's possible to simply adapt [`deriveEq`](https://hackage.haskell.org/package/deriving-compat-0.6.1/docs/Data-Eq-Deriving-Internal.html#v:deriveEq) without much change? It relies on [`buildTypeInstance`](https://hackage.haskell.org/package/deriving-compat-0.6.1/docs/src/Data.Deriving.Internal.html#buildTypeInstance), which accepts any [`ClassRep`](https://hackage.haskell.org/package/deriving-compat-0.6.1/docs/src/Data.Deriving.Internal.html#ClassRep), which is where we probably can fit our own `Eq` analogously to how they do it [here](https://hackage.haskell.org/package/deriving-compat-0.6.1/docs/src/Data.Eq.Deriving.Internal.html#Eq). No idea if it's actually going to work, though.

**Comment by uhbif19** [comment]
@effectfully Yes, there is ~100LOC module with `deriveEq` function in https://github.com/Liqwid-Labs/plutus-extra, and I will try to make PR with it on this week.

**Comment by effectfully** [comment]
Thank you very much!

**Comment by Unisay** [comment]
Closing this issue in favor of #7430, which tracks the same feature request (TH deriving for PlutusTx.Eq).
